File: backend/src/main/java/com/company/Entities/Video.java
package com.company.Entities;

public class Video {
    private Integer videoId, userId, views;
    private long uploadDate;
    private String videoURL, title, description, postedByUsername, likes, dislikes, stars;

    public Video(Integer videoId, Integer userId, long uploadDate, String videoURL, String title, String description, Integer views, String postedByUsername, String likes, String dislikes, String stars) {
        this.videoId = videoId;
        this.userId = userId;
        this.uploadDate = uploadDate;
        this.videoURL = videoURL;
        this.title = title;
        this.description = description;
        this.views = views;
        this.postedByUsername = postedByUsername;
        this.likes = likes;
        this.dislikes = dislikes;
        this.stars = stars;
    }

    public Video(Integer videoId, String title, String postedByUsername, String videoURL){
        this.videoId = videoId;
        this.videoURL = videoURL;
        this.title = title;
        this.postedByUsername = postedByUsername;
    }

    public Integer getVideoId() {
        return videoId;
    }

    public void setVideoId(Integer videoId) {
        this.videoId = videoId;
    }

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public Integer getViews() {
        return views;
    }

    public void setViews(Integer views) {
        this.views = views;
    }

    public long getUploadDate() {
        return uploadDate;
    }

    public void setUploadDate(long uploadDate) {
        this.uploadDate = uploadDate;
    }

    public String getVideoURL() {
        return videoURL;
    }

    public void setVideoURL(String videoURL) {
        this.videoURL = videoURL;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getPostedByUsername() {
        return postedByUsername;
    }

    public void setPostedByUsername(String postedByUsername) {
        this.postedByUsername = postedByUsername;
    }

    public String getLikes() {
        return likes;
    }

    public void setLikes(String likes) {
        this.likes = likes;
    }

    public String getDislikes() {
        return dislikes;
    }

    public void setDislikes(String dislikes) {
        this.dislikes = dislikes;
    }

    public String getStars() {
        return stars;
    }

    public void setStars(String stars) {
        this.stars = stars;
    }

    @Override
    public String toString() {
        return "Video{" +
                "videoId=" + videoId +
                ", userId=" + userId +
                ", views=" + views +
                ", uploadDate=" + uploadDate +
                ", videoURL='" + videoURL + '\'' +
                ", title='" + title + '\'' +
                ", description='" + description + '\'' +
                ", postedByUsername='" + postedByUsername + '\'' +
                ", likes='" + likes + '\'' +
                ", dislikes='" + dislikes + '\'' +
                ", stars='" + stars + '\'' +
                '}';
    }
}

File: frontend/src/jsClasses/general/Video.js
export default class Video {
  constructor(
    videoId,
    userId,
    uploadDate,
    videoURL,
    title,
    description,
    views,
    postedByUsername,
    likes,
    dislikes,
    stars
  ) {
    this.videoId = videoId;
    this.userId = userId;
    this.uploadDate = uploadDate;
    this.videoURL = videoURL;
    this.title = title;
    this.description = description;
    this.views = views;
    this.postedByUsername = postedByUsername;
    this.likes = likes;
    this.dislikes = dislikes;
    this.stars = stars;
  }

  getUploadDate() {
    return this.uploadDate;
  }

  setUploadDate(uploadDate) {
    this.uploadDate = uploadDate;
  }

  getStars() {
    return this.stars;
  }

  setStars(stars) {
    this.stars = stars;
  }

  getDislikes() {
    return this.dislikes;
  }

  setDislikes(dislikes) {
    this.dislikes = dislikes;
  }

  getLikes() {
    return this.likes;
  }

  setLikes(likes) {
    this.likes = likes;
  }

  getPostedByUsername() {
    return this.postedByUsername;
  }

  setPostedByUsername(username) {
    this.postedByUsername = username;
  }

  getVideoId() {
    return this.videoId;
  }

  setVideoId(videoId) {
    this.videoId = videoId;
  }

  getuserId() {
    return this.userId;
  }
  setuserId(userId) {
    this.userId = userId;
  }

  getVideoURL() {
    return this.videoURL;
  }

  setVideoUrl(videoURL) {
    this.videoURL = videoURL;
  }

  getTitle() {
    return this.title;
  }

  setTitle(title) {
    this.title = title;
  }

  getDescription() {
    return this.description;
  }

  setDescription(description) {
    this.description = description;
  }

  getViews() {
    return this.views;
  }

  setViews(views) {
    this.views = views;
  }
}
